internal/host/common.go

```diff
@@ -102,20 +102,16 @@ func GetHostnameAndRoleByIP(ip string, hosts []*models.Host) (string, models.Hos
 		}
 		inv, err := hostutil.UnmarshalInventory(h.Inventory)
 		if err != nil {
-			return "", "", fmt.Errorf("unable to unmarshall cluster inventory for host %s", h.RequestedHostname)
-		}
-		if strings.Count(ip, ":") == 0 {
-			for _, i := range inv.Interfaces {
-				for _, ipv4 := range i.IPV4Addresses {
-					if ip == ipv4 {
-						return getRealHostname(h, inv), h.Role, nil
-					}
-				}
-			}
+			return "", "", fmt.Errorf("unable to unmarshall cluster inventory for host %s: %s", h.RequestedHostname, err)
 		}
 		for _, i := range inv.Interfaces {
-			for _, ipv6 := range i.IPV6Addresses {
-				if ip == ipv6 {
+			ips := append(i.IPV4Addresses, i.IPV6Addresses...)
```

Comment:
Please add a UT that shows that this logic is working correctly in all cases

Reply:
Will this be sufficient?
https://play.golang.org/p/OYhsH6TkKR-

Reply:
This is not a unit test
will it run anc check the logic everytime someone makes a change?

Reply:
This is not a unit test
will it run anc check the logic everytime someone makes a change?

Reply:
Ok, I'll add unit test for this